### PR TITLE
.NET change

### PR DESCRIPTION
**Pull request by emveepee**
Changes for local run and API change

### CODE REVIEW

File: dotnet/trunk/LivePvrDataDotNet/Client.cs [hunk 1]
```diff
@@ -129,7 +129,7 @@ public Response GetStatus(StatusRequest req)
                 string payload = serializer.Serialize(req);
                 WebClient clnt = GetWebClient();
                 SignRequest(clnt, payload);
-                payload = clnt.UploadString(uri + "api/qryStatus", "q=" + HttpUtility.UrlEncode(serializer.Serialize(req)));
+                payload = clnt.UploadString(uri + "livepvrdata/query", "q=" + HttpUtility.UrlEncode(serializer.Serialize(req)));
```
Comment (Slugger) [inline dotnet/trunk/LivePvrDataDotNet/Client.cs]: You're assuming the deployed app context, which I don't think is correct.  It should be just `query`, which is relative and should automatically deduce the deployed context based on the base uri being used.  You could probably mod the default uri to be `localhost:8080/livepvrdata` if the default context is livepvrdata (I don't remember if that's the case).  Either way, hardcoding the app context here is not a great idea.

**Review by Slugger — CHANGES_REQUESTED** [review]
Hardcoding of app context not best idea.

**Comment by emveepee** [comment]
If I change it to query, then  the user has to pass

http://ip:port/livepvrdata to override the default.  That seems confusing to me.

**Comment by Slugger** [comment]
The problem with hardcoding the context here is that your client is no good -- and can't be reconfigured at runtime -- for use against a server that doesn't deploy to the hardcoded context you've specified here.

For example, when I ran the service on Google, I just ran it from the default context, which is why the default url was just `www.livepvrdata.com/` and to do a query you'd have to hit `www.livepvrdata.com/query`.  With your change here, you're assuming every instance of the service will always deploy to a context named `livepvrdata` and if a service deployment doesn't, your client is no good and will never be able to query it.  Assuming an app context is never a good idea.

**Comment by Slugger** [comment]
But, as I said, if you want to include `livepvrdata` as the default context in the `DEFAULT_URI` then that would be fine.  And that probably even makes sense since I'm assuming you did this because the service will deploy to `livepvrdata` by default.

**Comment by emveepee** [comment]
Not including the context in the default was my concern confusion, I have modified the pull request as suggested.  Thanks

**Review by Slugger — APPROVED** [review]

**Comment by Slugger** [comment]
Merged this PR.  I assume you tested the changes, my naked eye thinks there might be a missing slash in the url when using the `DEFAULT_URI` as the base uri, but if you tested it then I'm wrong and that's fine.  If I happen to be right, you'll find out soon enough and can just submit another PR. ;)

**Comment by emveepee** [comment]
Yes there is a missing / because my code overrides the default.    I was just going to do an update.

Martin

**Pull request merged**